src/service/logs/ingest.rs

```diff
@@ -274,19 +266,26 @@ pub async fn ingest(
             }
         };
 
-        let function_no = stream_before_functions_map
-            .get(&main_stream_key)
-            .map(|v| v.len())
-            .unwrap_or_default()
-            + stream_after_functions_map
-                .get(&key)
-                .map(|v| v.len())
-                .unwrap_or_default();
-        let (ts_data, fn_num) = json_data_by_stream
-            .entry(routed_stream_name.clone())
-            .or_insert((Vec::new(), None));
-        ts_data.push((timestamp, local_val));
-        *fn_num = need_usage_report.then_some(function_no);
+        // Refactor and assign data to each destination stream.
+        for dest in destination_stream_names {
+            // The data for each of its destination streams since the data may
+            // need to be assigned to multiple streams and may to be refactored
+            // differently to match the user-defined schema of that stream.
+            let local_val_for_dest = local_val.clone();
+
+            // Refactor the data to match the user.defined schema of the
+            // destination stream.
+            let local_val_for_dest = if let Some(fields) = user_defined_schema_map.get(&dest) {
+                crate::service::logs::refactor_map(local_val_for_dest, fields)
+            } else {
+                local_val_for_dest
+            };
+
+            let (ts_data, _) = json_data_by_stream
+                .entry(dest)
+                .or_insert((Vec::new(), None));
+            ts_data.push((timestamp, local_val_for_dest));
```

Comment:
As mentioned in a [previous comment](https://github.com/openobserve/openobserve/pull/4366/files#r1734710776), here is where we update the assigned data to each destination stream. We don't need to update the function usage count here because that data has already been updated (specifically for the original stream name key) when the functions were originally applied.